crates/biome_service/src/configuration/parse/json/vcs.rs

```diff
@@ -64,9 +69,9 @@ impl DeserializationVisitor for VcsConfigurationVisitor {
         }
         if result.client_kind.is_none() && !result.is_disabled() {
             diagnostics.push(
-                DeserializationDiagnostic::new(markup! {
-                    "You enabled the VCS integration, but you didn't specify a client."
-                })
+                DeserializationDiagnostic::new(
+                    "You enabled the VCS integration, but you didn't specify a client.",
+                )
```

Comment:
The check here is an example of a use case that could be solved through a `DeserializableValidator` trait. If a struct contains a `#[deserializable(with_validator)]` annotation, we could let the generated deserializer call into a `validate()` function that would have the ability to reject the deserialized instance. But I haven't built that yet :)

Reply:
Did you notice other code that requires a validation?

Reply:
Yeah, for instance `LineWidth` which checks whether the `u16` is within its own `MIN` and `MAX`. And also a lot of the types in the generated ` rules.rs` have a similar kind of validation logic.